dealer.go

```diff
@@ -108,19 +113,28 @@ func (d *Dealer) ReceiveMessage(sessionID int64, msg messages.Message) (*Message
 			break
 		}
 
+		receiveProgress, ok := call.Options()[OptionReceiveProgress].(bool)
+		if !ok {
```

Comment:
this check can be removed and the value of `receiveProgress` maybe used directly without reassining.
```suggestion
		if !ok {
```